forte/data/extractor/extractor.py

```diff
@@ -0,0 +1,230 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import codecs
+import logging
+import os
+import numpy as np
+import torch
+from torch import Tensor
+from typing import Iterator, Dict, List, Any, Union, Set, Tuple
+
+from forte.data.data_pack import DataPack
+from forte.data.data_utils_io import dataset_path_iterator
+from forte.data.readers.base_reader import PackReader
+from ft.onto.base_ontology import Token, Sentence, Document, Annotation, EntityMention
+from forte.data.ontology.core import EntryType
+from forte.data.span import Span
+from forte.data.extractor.vocabulary import Vocabulary
+from forte.data.extractor.feature import Feature
+
+
+class BaseExtractor:
+    def __init__(self, config: Dict):
+        self.config = config
+        self.entry = config["entry"]
+        use_pad = config.get("vocab_use_pad", True)
+        use_unk = config.get("vocab_use_unk", False)
+        method = config.get("vocab_method", "indexing")
+        self.__vocab = Vocabulary(method = method,
+                                use_pad = use_pad,
+                                use_unk = use_unk)
+
+    # Wrapper functions for vocabulary class,
+    # so that vocab is not directly exposed to
+    # outside user.
+    def size(self):
```

Comment:
If I only use vocabulary within extractor, I think I don't need to do this wrapper. But one case I have discuss with Yanwen is that, if the user want to add some entry or access the size of the vocab from outside the extractor, it might not look good if he needs to call "extractor.vocab.add_entry" or "extractor.vocab.size". 
So, I am not sure do we need to wrap functions of the vocab.